packages/react-charting/src/components/CommonComponents/ErrorBoundary.base.tsx

```diff
@@ -0,0 +1,107 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import * as React from 'react';
+import { classNamesFunction, getId } from '@fluentui/react/lib/Utilities';
+import { IErrorBoundaryProps, IErrorBoundaryStyles } from './ErrorBoundary.types';
+import { ErrorBadgeIcon, ErrorIcon, StatusErrorFullIcon } from '@fluentui/react-icons-mdl2';
+
+export enum ErrorCodes {
+  NoData = 'No data',
+  GeneralError = 'General error',
+}
+
+interface IErrorBoundaryState {
+  hasError: boolean;
+  errorMsg: string;
+}
+const getClassNames = classNamesFunction<IErrorBoundaryProps, IErrorBoundaryStyles>();
+
+export class ErrorBoundaryBase extends React.Component<IErrorBoundaryProps, IErrorBoundaryState> {
+  private _defaultErrorMsg = "Something went wrong and we couldn't get the page to display";
+  private _classNames;
+  public static getDerivedStateFromError(error: Error) {
+    return { hasError: true, errorMsg: error.message };
+  }
+
+  constructor(props: IErrorBoundaryProps) {
+    super(props);
+    this.state = {
+      hasError: false,
+      errorMsg: '',
+    };
+    this._classNames = getClassNames(this.props.styles!, {
+      hasEmptyState: this.props.hasEmptyState,
+      theme: this.props.theme!,
+      width: this.props.width,
+      height: this.props.height,
+    });
+  }
+
+  public componentDidCatch(_error: any, errorInfo: any) {
+    this.setState({
+      hasError: true,
+      errorMsg: errorInfo,
+    });
+  }
+
+  public render() {
+    if (this.state.hasError || this.props.hasEmptyState) {
+      if (this.state.hasError && this.props.handleError !== undefined) {
+        return this.props.handleError();
+      }
+
+      if (this.props.hasEmptyState && this.props.handleEmptyState !== undefined) {
```

Comment:
Log a task to move all empty data handling logic in all charts to this common control.

Reply:
Since the checking for empty data is different for different charts, I don't think the moving of empty data handling logic in a generic way will be possible. The checking will depend on individual chart data types. For example, for Area chart, we do the following:
private _isChartEmpty(): boolean {
    return !(
      (
        this.props.data &&
        this.props.data.lineChartData &&
        this.props.data.lineChartData.length > 0 &&
        this.props.data.lineChartData.filter(item => item.data.length === 0).length === 0
      )
      // if all the data sets have no data
      // filtering all items which have no data and checking if the length of the filtered array is 0
      // which means chart is not empty
    );
  }

while for Vertical bar chart, we do the following:
private _isChartEmpty(): boolean {
    return (
      this._points.length === 0 ||
      (d3Max(this._points, (point: IVerticalBarChartDataPoint) => point.y)! <= 0 && !this._isHavingLine)
    );
  }

Reply:
you can have a callback reporting whether the data is empty or not.

Reply:
have done that, added an error-utils which checks